feat: support Celery auto-scale

ZuzooVn opened the pull request:
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Support Celery auto-scale

Fixes 

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)

ZuzooVn committed:
chore: update i18n for #5811

ZuzooVn committed:
Merge branch 'langgenius:main' into main

ZuzooVn committed:
Merge branch 'main' of https://github.com/ZuzooVn/dify

ZuzooVn committed:
fix: i18n for #5943

ZuzooVn committed:
chore: update i18n for #6069

ZuzooVn committed:
Merge branch 'langgenius:main' into main

ZuzooVn committed:
Merge branch 'main' of https://github.com/ZuzooVn/dify

ZuzooVn committed:
feat: support celery autoscale

bowenliang123 commented on api/docker/entrypoint.sh:
```diff
@@ -8,8 +8,16 @@ if [[ "${MIGRATION_ENABLED}" == "true" ]]; then
-  celery -A app.celery worker -P ${CELERY_WORKER_CLASS:-gevent} -c ${CELERY_WORKER_AMOUNT:-1} --loglevel INFO \
-    -Q ${CELERY_QUEUES:-dataset,generation,mail,ops_trace,app_deletion}
+  if [ "${CELERY_AUTO_SCALE,,}" = "true" ]; then
+    MAX_WORKERS=${CELERY_MAX_WORKERS:-4}
```
We could set the max workers by using `nproc` to identify the number available CPU cores, as celery worker command itself suggested.

ZuzooVn committed:
feat: auto get the number of available CPU cores

ZuzooVn committed:
chore: support --concurrency

ZuzooVn committed:
Merge branch 'main' into celery-worker-auto-scale

takatost commented on api/docker/entrypoint.sh:
```diff
@@ -8,8 +8,26 @@ if [[ "${MIGRATION_ENABLED}" == "true" ]]; then
+    CONCURRENCY_OPTION="-c ${CELERY_WORKER_AMOUNT}"
+  else
+    # Otherwise, the default is the number of CPUs available on the system
+    CONCURRENCY_OPTION=""
```
I think it should be set to 1 even if CELERY_WORKER_AMOUNT is not set, otherwise users will automatically start multiple workers after upgrading to the next version, resulting in insufficient memory resources and OOM.

ZuzooVn committed:
set to 1 workers even if CELERY_WORKER_AMOUNT is not set

otherwise users will automatically start multiple workers after upgrading to the next version, resulting in insufficient memory resources and OOM

takatost committed:
chore: optimize entrypoint.sh

takatost reviewed (approved):
thanks
